### ISSUE TITLE
Literal search onboarding for new instances and users

### ISSUE THREAD

**Issue by nicksnyder** [sourcegraph/sourcegraph#6579]
Right now we have this alert that shows up for all users:

![Screen Shot 2019-11-13 at 11 06 56 AM](https://user-images.githubusercontent.com/754768/68795985-dddc6380-0606-11ea-8a96-976879bf39e5.png)

We created this alert primarily to educate existing users on the change to default behavior, but it shows for all users because we don't currently have a good technical solution for detecting the difference between new and old instances.

There is a product question of whether we should display this new instances and new users. @sqs seems to think no, and @slimsag seems to think yes. @christinaforney thoughts?

Engineering question, do we foresee wanting to make this distinction in the future? I think the answer is yes. It would be nice to have the capability to communicate in-product changes directly to the set of affected users (instead of all users). One potential solution for this would be to add a DB table that tracks the current version of Sourcegraph. Two columns: version_number and inserted_at

Pseudocode that is executed on startup of a Sourcegraph instance
```
select version_number, inserted_at from versions order by inserted_at desc limit 1
if version_number == current_version_number_of_binary
    return // do nothing
else
    insert into versions (current_version_number, now())
```

Any other code path can check if the instance is "new" with `select count(*) from versions`. If there is only one version, then the instance is "new".

**Comment by attfarhan** [comment]
It would be nice to have this distinction. I can remember there was one other instance a while back where I needed to use release dates.

I can own this, but it still wouldn't solve the issue immediately, since we'll only have one entry in the table when customers upgrade.

**Comment by nicksnyder** [comment]
Yes, the technical implementation would only benefit future use cases. We still need a product decision for the immediate issue with literal search.

**Comment by christinaforney** [comment]
### Adding ability to distinguish between new/upgraded instances
Agree that distinguishing between a new instance and an existing instance (upgrade path vs new path) will be important to us as we continue to make improvements. Is there a separate ticket tracking that? This seems to outline the potential solution, but the question is targeted at the literal search toast.

### Literal search alert
I have not heard any user or customer complaints about the new experience being confusing. I think there has been more confusion with the toast than the actual feature. To find a balance, I propose leaving the alert in for the 3.10 release, and removing it after. Because we only support +2 upgrades, having it in 3.9 and 3.10 means that no matter what, every instance will have to go to a version that has the alert. 

Thoughts?

**Comment by nicksnyder** [comment]
> every instance will have to go to a version that has the alert.

This doesn't necessarily mean that every user will actually see the alert. If I am a customer on 3.7 and I want to upgrade to 3.11, I will have to upgrade to 3.9 first, but then I can immediately upgrade to 3.11 in the same downtime.

**Comment by attfarhan** [comment]
Added an issue for the db table: https://github.com/sourcegraph/sourcegraph/issues/6643.

> I have not heard any user or customer complaints about the new experience being confusing
If this is the case, I am OK with removing it in 3.12 or later.

@nicksnyder is correct that it's possible not all users will see the toast, which is a downside of removing it in 3.11. 

If we get the DB table in 3.11, by 3.12, we can add logic saying to only show the toast to users who have upgraded. This won't capture all users that haven't seen the toast, since there will be instances upgrading from before 3.11, that won't have multiple entries in the DB table, and therefore won't see the message. But, I think we can feel a bit safer that those users were at least on 3.9 and have seen the notification before. This means by 3.12, we are absolutely not showing the toast to new users.

Then, we can remove the toast completely in 3.13.

**Comment by attfarhan** [comment]
If my above solution is waiting too long to remove the toast for new users, I am OK with removing it in 3.11 too, based on my expectation that regexp-search users read our docs and release announcements.
